src/codegen_arc_gc.cpp

```
@@ -0,0 +1,353 @@
+    // Standard cycle detection: white/grey/black colouring.
+    enum Color { White, Grey, Black };
+    std::unordered_map<std::string, Color> color;
+    for (auto &t : all_types) color[t] = White;
+
+    std::unordered_set<std::string> cyclic;
+
+    // DFS that returns true if the node is part of or reaches a cycle.
+    std::function<bool(const std::string &, std::vector<std::string> &)> dfs;
+    dfs = [&](const std::string &node, std::vector<std::string> &stack) -> bool {
+        color[node] = Grey;
+        stack.push_back(node);
```

Cppcheck (2.20.0)

[style] The function 'hasDirective' is never used.

(unusedFunction)